src/snowflake/snowpark/column.py

```diff
@@ -250,6 +250,7 @@ class Column:
     #       For example, running: df.filter(col("A").isin(1, 2, 3) & col("B")) would fail since the boolean operator
     #       '&' would try to construct an AST using that of the new col("A").isin(1, 2, 3) column (which we currently
     #       don't fill if the only argument provided in the Column constructor is 'expr1' of type Expression)
```

Comment:
I recall there used to be some issue adding this to the constructor, do we have tests covering on/off for using `Column` objects (i.e. calling this API, and also `col`)?

Reply:
Could you clarify, do you mean on/off in terms of our global AST enabled flag?

Reply:
oh like with AST flag enabled/disabled.

Reply:
Oh, yes we do have tests in both cases then (public calls made by tests, and internal calls to the constructor) if I'm understanding correctly.